satellite/audit/reservoir.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"math/rand"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// Reservoir holds a certain number of segments to reflect a random sample
+type Reservoir struct {
+	Paths []storj.Path
+	Size  int
+	index int
+}
+
+// NewReservoir instantiates a Reservoir
+func NewReservoir(size int) *Reservoir {
+	return &Reservoir{
+		Size:  size,
+		index: 0,
+	}
+}
+
+// Sample makes sure that for every segment in metainfo from index i=size..n-1,
+// pick a random number r = rand(0..i), and if r < size, replace reservoir.Segments[r] with segment
+func (reservoir *Reservoir) Sample(path storj.Path) {
+	reservoir.index++
```

Comment:
nit - I think `numSeen` or `seenCount` or something makes more sense than `index` for this variable.

Reply:
you will have to take that up with @kaloyan-raev :) https://github.com/storj/storj/pull/2744#discussion_r312808764

Reply:
<.<